onchain/src/tests/art_peace.cairo

```diff
@@ -425,3 +425,209 @@ fn deposit_reward_test() {
+    let first_vote = 1;
+    let second_vote = 2;
+
+    snf::start_prank(CheatTarget::One(art_peace_address), utils::PLAYER1());
+    art_peace.vote_color(first_vote);
+    snf::stop_prank(CheatTarget::One(art_peace_address));
+    assert_eq!(art_peace.get_color_votes(first_vote), 1, "Wrong number of votes");
+    assert_eq!(art_peace.get_color_votes(second_vote), 0, "Wrong number of votes");
+
+    snf::start_prank(CheatTarget::One(art_peace_address), utils::PLAYER1());
+    art_peace.vote_color(second_vote);
+    snf::stop_prank(CheatTarget::One(art_peace_address));
+    assert_eq!(art_peace.get_color_votes(first_vote), 0, "Wrong number of votes");
+    assert_eq!(art_peace.get_color_votes(second_vote), 1, "Wrong number of votes");
+}
+
+#[test]
+fn get_votable_colors_test() {
+    let art_peace_address = deploy_contract();
+    let art_peace = IArtPeaceDispatcher { contract_address: art_peace_address };
+    let votable_colors = art_peace.get_votable_colors();
+    let expected_colors = array![
+        0xDD0000,
+        0x00DD00,
+        0x0000DD,
+        0xDDDD00,
+        0xDD00DD,
+        0x00DDDD,
+        0x880000,
+        0x008800,
+        0x000088,
+        0x888800,
+    ];
+    assert_eq!(votable_colors.len(), expected_colors.len(), "Wrong number of votable colors");
+    assert_eq!(votable_colors, expected_colors, "Wrong votable colors");
+}
+
+#[test]
+fn votable_colors_is_updated_according_to_votes_test() {
+    let art_peace_address = deploy_contract();
+    let art_peace = IArtPeaceDispatcher { contract_address: art_peace_address };
+    let vote_player1 = 2; // 0x00DD00,
+    let vote_player2 = 3; // 0x0000DD,
+    let vote_player3 = 4; // 0xDDDD00,
+
+    let expected_colors = array![
+        0xDD0000, // 0x00DD00,
+         // 0x0000DD,
+        // 0xDDDD00,
+        0xDD00DD, 0x00DDDD, 0x880000, 0x008800, 0x000088, 0x888800,
+    ];
+
+    snf::start_prank(CheatTarget::One(art_peace_address), utils::PLAYER1());
+    art_peace.vote_color(vote_player1);
+    snf::stop_prank(CheatTarget::One(art_peace_address));
+
+    snf::start_prank(CheatTarget::One(art_peace_address), utils::PLAYER2());
+    art_peace.vote_color(vote_player2);
+    snf::stop_prank(CheatTarget::One(art_peace_address));
+
+    snf::start_prank(CheatTarget::One(art_peace_address), utils::PLAYER3());
+    art_peace.vote_color(vote_player3);
+    snf::stop_prank(CheatTarget::One(art_peace_address));
+
+    let old_votable_colors_count = art_peace.get_votable_colors().len();
+    snf::start_warp(CheatTarget::One(art_peace_address), DAY_IN_SECONDS);
+    art_peace.increase_day_index();
+    let votable_colors = art_peace.get_votable_colors();
+    assert_eq!(
+        votable_colors.len() + 3, old_votable_colors_count, "Number of votable colors not updated"
+    );
+    assert_eq!(votable_colors, expected_colors, "Votable colors mismatch");
+    snf::stop_warp(CheatTarget::One(art_peace_address));
+}
+
```

Comment:
Nice tests! Thank you!

I think adding a test where you give a bunch of colors various votes and test to make sure things work in weird cases like I mentioned w/ 
Color 1 : 5 votes
Color 2 : 8 votes
Color 3 : 7 votes
Color 4 : 5 votes
Color 5 : 4 votes

That would be cool. You should be able to use snf::store to just update the values in `color_votes` without needing to create a bunch of players to vote with ( or you could create a custom util function to just use a bunch of constant addresses to vote with ).

Reply:
Ok I have added an helper function to easily setup color votes:
```cairo
fn set_vote(day: u32, color_index: u8, vote: u32, contract_address: ContractAddress) {
    let mut state = ArtPeace::contract_state_for_testing();
    let storage_address: felt252 = storage_address_from_base(state.color_votes.address((color_index, day))).into();
    let mut storage_value: Array<felt252> = ArrayTrait::new();
    storage_value.append(vote.into());
    snf::store(contract_address, storage_address, storage_value.span());
}
```